### PR TITLE
fix(local-dev): postgres connection max age to 0?

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Developing locally, whenever I save a frontend source file, the app reloads. Django keeps its postgres connection strings around for 10 minutes, and reuses a new one every time the app reloads.

I get very easily into a state where I need to restart django to free up all the blocked DB connections.

image

## Changes

Changes the django postgres connection max age to 0 when in `DEBUG`. This frees the connection immediately. 

## How did you test this code?

Ran with this for a while locally, didn't see the app crash anymore.

## Questions

I don't know the backstory behind this parameter, and what kind of implications this has for production. I'm always weary of situations where local dev differs from production, and this seems to be one of those cases. 

I could add this as yet another env that you can configure locally, but I'm just curious how is it that I'm the only one experiencing this. Am I?

### CODE REVIEW

**Comment by mariusandra** [comment]
It's really impossible to develop locally with this setting set. I removed it totally, since I assume we're using pgbouncer on cloud and it does its own pooling anyway.

File: posthog/settings/data_stores.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
     DATABASE_URL = os.getenv("DATABASE_URL", "")
 
 if DATABASE_URL:
-    DATABASES = {"default": dj_database_url.config(default=DATABASE_URL, conn_max_age=600)}
+    DATABASES = {"default": dj_database_url.config(default=DATABASE_URL, conn_max_age=0 if DEBUG else 600)}
```
Comment (neilkakkar) [inline posthog/settings/data_stores.py:34]:
```suggestion
    DATABASES = {"default": dj_database_url.config(default=DATABASE_URL, conn_max_age=0)}
```

**Comment by neilkakkar** [comment]
Follow up on this, @ellie I checked our prod env vars, and it seems like we don't use this conn_max_age in prod at all. It's set to 0, because we don't pass direct `DATABASE_URL` param in prod. It seems like this is only used in tests anyway 😅 .

So, I think we can just make the default 0 as well? (which is also what it is currently in prod)

**Comment by ellie** [comment]
> Follow up on this, @ellie I checked our prod env vars, and it seems like we don't use this conn_max_age in prod at all. It's set to 0, because we don't pass direct `DATABASE_URL` param in prod. It seems like this is only used in tests anyway 😅 .
> 
> So, I think we can just make the default 0 as well? (which is also what it is currently in prod)

Sure, makes sense to me!

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**